source/progress.ts

```diff
@@ -75,8 +71,7 @@ export function uploadProgress(request: ClientRequest, emitter: EventEmitter, up
 		const onSocketConnect = (): void => {
 			progressInterval = setInterval(() => {
 				const lastUploadedBytes = uploadedBytes;
-				/* istanbul ignore next: see #490 (occurs randomly!) */
-				const headersSize = (request as any)._header ? Buffer.byteLength((request as any)._header) : 0;
+				const headersSize = Buffer.byteLength((request as any)._header);
```

Comment:
Yup, double checked that. Is always initialized on `request.end()`. I couldn't find any code that would overwrite it ;)

Reply:
But you can't guarantee that won't change in a future Node.js version. I'm being paranoid as it's an internal variable (the `_` prefix), so I think it would be smart to stay on the safe side here.

Reply:
But then it would fail silently. It's better to fail loudly IMO

Reply:
So do a `console.warn` then. Crashing hard over an implementation detail is not a good experience. Generally, when using private APIs, you're expected to properly guard it. Node.js is free to change it in a patch version and it will affect people before we can get to patch it. It's better to just return a dummy value and warn than crashing hard.

Reply:
will do